### PR TITLE
[PATCH v3] Merge ODP linux-generic v1.44.0.0

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

File: platform/linux-dpdk/include/odp_packet_internal.h [hunk 1]
```diff
@@ -218,9 +218,11 @@ static inline struct rte_mbuf *pkt_to_mbuf(odp_packet_t  pkt)
 	return (struct rte_mbuf *)(uintptr_t)pkt;
 }
 
-static inline void packet_subtype_set(odp_packet_t pkt, int ev)
+static inline void packet_subtype_set(odp_packet_t pkt, int subtype)
 {
-	packet_hdr(pkt)->event_hdr.subtype = ev;
+	odp_packet_hdr_t *pkt_hdr = packet_hdr(pkt);
+
+	pkt_hdr->event_hdr.subtype = subtype;
```
Comment (TuomasTaipale) [inline platform/linux-dpdk/include/odp_packet_internal.h:225]: Not sure why this couldn't still remain as a oneliner?

File: platform/linux-dpdk/include/odp_packet_internal.h [hunk 2]
```diff
@@ -236,7 +238,9 @@ static inline void packet_init(odp_packet_hdr_t *pkt_hdr, odp_pktio_t input)
 	pkt_hdr->p.l3_offset        = ODP_PACKET_OFFSET_INVALID;
 	pkt_hdr->p.l4_offset        = ODP_PACKET_OFFSET_INVALID;
 
-	pkt_hdr->event_hdr.subtype = ODP_EVENT_PACKET_BASIC;
+	if (odp_unlikely(pkt_hdr->event_hdr.subtype != ODP_EVENT_PACKET_BASIC))
+		pkt_hdr->event_hdr.subtype = ODP_EVENT_PACKET_BASIC;
+
```
Comment (TuomasTaipale) [inline platform/linux-dpdk/include/odp_packet_internal.h:243]: Is anything achieved with the extra check? The field is anyways already accessed in the conditional check.

File: platform/linux-dpdk/include/odp_timer_internal.h [hunk 3]
```diff
@@ -56,10 +58,12 @@ ODP_STATIC_ASSERT(sizeof(odp_timeout_hdr_t) <= 3 * RTE_CACHE_LINE_SIZE,
 void _odp_timer_run_inline(int dec);
 
 /* Static inline wrapper to minimize modification of schedulers. */
-static inline void timer_run(int dec)
+static inline uint64_t timer_run(int dec)
 {
 	if (odp_global_rw->inline_timers)
 		_odp_timer_run_inline(dec);
+
```
Comment (TuomasTaipale) [inline platform/linux-dpdk/include/odp_timer_internal.h:71]: Perhaps the unavailability of next timeout could be commented here.

**Review by TuomasTaipale — COMMENTED** [review]
A few minor comments.

File: platform/linux-dpdk/include/odp_packet_internal.h (hunk 1 above)
Comment (MatiasElo) [reply platform/linux-dpdk/include/odp_packet_internal.h]: Was just mechanical copy-paste. Updated in V2.

File: platform/linux-dpdk/include/odp_packet_internal.h (hunk 2 above)
Comment (MatiasElo) [reply platform/linux-dpdk/include/odp_packet_internal.h:241]: Making the write conditional provides ~3-4% performance increase in odp_pool_perf test on my system.

File: platform/linux-dpdk/include/odp_timer_internal.h (hunk 3 above)
Comment (MatiasElo) [reply platform/linux-dpdk/include/odp_timer_internal.h:71]: Added comment in V2.

**Comment by MatiasElo** [comment]
V2:
- Fixed review comments
- Documented that DPDK v23.11 is now supporter and added it to CI

File: platform/linux-dpdk/include/odp_packet_internal.h (hunk 2 above)
Comment (TuomasTaipale) [reply platform/linux-dpdk/include/odp_packet_internal.h:241]: There might be a lot more variability in the subtype with an actual application, I wonder how much accessing the branch then in turn slows down. Anyways, academic, resolving this.

**Review by TuomasTaipale — APPROVED** [review]

**Pull request merged**